### DISCUSSION TITLE
WebXR not working in Samsung Internet browser?

### DISCUSSION THREAD

**Discussion by ronpadz** [google/model-viewer#3364] [Q&A]
Just installed and tested the latest Samsung Internet browser on my Pixel 6. After granting permission to the camera, the camera indicator light shows green yet it doesn't open. When closing out of WebXR, the camera becomes visible for a brief moment.

Anyone else experiencing this?

https://user-images.githubusercontent.com/32022543/163257070-b58b464e-b604-4fae-bd06-e2520a8892ef.mp4

**Comment by milesgreen** [comment]
It appears to be only in Dark Mode that the WebXR AR session is not displayed - no camera feed no 3D content. When Dark Mode is off, WebXR content is displayed correctly. This appears universal to all WebXR content, not specific to Model-Viewer.

Issue has been reported at the the Samsung Developer Forum:
https://forum.developer.samsung.com/t/dark-mode-prevents-webxr-ar-session-from-displaying-any-content/19868

**Reply by milesgreen** [reply]
@AdaRoseCannon - this might be of interest to you?

**Comment by AdaRoseCannon** [comment]
This is a known bug. What version of Samsung Internet are you using? Is it still broken in the current Beta? it's supposed to be fixed in recent versions.

**Comment by milesgreen** [comment]
Thanks @AdaRoseCannon. I still see the error with both Samsung Internet public **16.2.5.4** and beta **17.0.1.65** on a Pixel 4 XL, Android 12.
Good to know it's being tracked.
Thanks

**Comment by milesgreen** [comment]
@AdaRoseCannon - just FYI - still seeing this as an issue on Samsung Internet 17.0.1.69

In addition, there's also a new issue with exiting a WebXR session blocking any page rendering:
https://forum.developer.samsung.com/t/webxr-critical-blocker-exiting-webxr-ar-session-leaves-browser-window-blank-unable-to-render-any-content/20152/3

**Reply by milesgreen** [reply]
As of Samsung Internet **17.0.6.9**, this blank window issue also seems to be resolved. Not sure if it was fixed in an earlier release? but definitely working in **17.0.6.9**. Great.

**Comment by milesgreen** [comment]
Testing again in Samsung Internet **17.0.6.9** and WebXR is no longer breaking when in dark mode. Might have been fixed in an earlier release? But this is the first time I've retested in a little while. Good stuff. 👍

**Comment by AdaRoseCannon** [comment]
I am glad to hear it
